tests/test_record_aggregator.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import unittest
+
+from model_analyzer.record.record_aggregator import RecordAggregator
+from model_analyzer.analyzer.perf_analyzer.perf_throughput import PerfThroughput
+from model_analyzer.analyzer.perf_analyzer.perf_latency import PerfLatency
+from model_analyzer.analyzer.perf_analyzer.perf_config import PerfAnalyzerConfig
+
+
+class TestRecordAggregatorMethods(unittest.TestCase):
+    def test_insert_record(self):
+        record_aggregator = RecordAggregator()
+
+        self.assertEqual(record_aggregator.total(), 0)
+
+        throughput_record = PerfThroughput(5, PerfAnalyzerConfig())
+        record_aggregator.insert_record(throughput_record)
+
+        # Assert record is added
+        self.assertEqual(record_aggregator.total(), 1)
```

Comment:
Test the value of the record it returns. I have no idea that the aggregator didn't change the value. Lock in the behavior that the record does not change.

Reply:
This is tested in `test_filter_records`